### PR TITLE
Update the module with latest syntax

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
Update the module with latest syntax.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,23 +48,25 @@ endpoint amazons3:Client amazonS3Client {
...
 import wso2/amazons3;
 
-function main(string... args) {
-    endpoint amazons3:Client amazonS3Client {
-        accessKeyId:"",
-        secretAccessKey:"",
-        region:""
+public function main() {
+    AmazonS3Configuration amazonS3Config = {
+    accessKeyId: testAccessKeyId,
```
Comment (ldclakmal) [inline README.md]: Please format code here.

File: amazons3/Module.md [hunk 2]
```diff
@@ -45,128 +45,125 @@ and bucketName in the Amazon S3 client config.
...
-    accessKeyId:"<your_access_key_id>",
-    secretAccessKey:"<your_secret_access_key>",
-    region:"<your_region>"
-};
+AmazonS3Configuration amazonS3Config = {
+    accessKeyId: testAccessKeyId,
+    secretAccessKey: testSecretAccessKey,
+    region: testRegion,
+    amazonHost: amazonHost
+    };
```
Comment (ldclakmal) [inline amazons3/Module.md]: Please format code here.

File: amazons3/Module.md [hunk 3]
```diff
@@ -45,128 +45,125 @@ and bucketName in the Amazon S3 client config.
...
 import wso2/amazons3;
 
-function main(string... args) {
-    endpoint amazons3:Client amazonS3Client {
-        accessKeyId:"<your_access_key_id>",
-        secretAccessKey:"<your_secret_access_key>",
-        region:"<your_region>"
+public function main() {
+    AmazonS3Configuration amazonS3Config = {
+    accessKeyId: "<your_access_key_id>",
```
Comment (ldclakmal) [inline amazons3/Module.md]: Please format code here.

File: amazons3/Module.md [hunk 4]
```diff
@@ -45,128 +45,125 @@ and bucketName in the Amazon S3 client config.
...
     var getBucketListResponse = amazonS3Client -> getBucketList();
-    match getBucketListResponse {
-        amazons3:Bucket[] buckets => {
-            io:println("Listing all buckets: ");
-            foreach bucket in buckets {
-                io:println("Bucket Name: " + bucket.name);
-            }
+    if (getBucketListResponse is amazons3:Bucket[]) {
+        io:println("Listing all buckets: ");
+        foreach bucket in getBucketListResponse {
```
Comment (ldclakmal) [inline amazons3/Module.md]:
```suggestion
        foreach var bucket in getBucketListResponse {
```

File: amazons3/Module.md [hunk 5]
```diff
@@ -45,128 +45,125 @@ and bucketName in the Amazon S3 client config.
...
-        amazons3:S3Object[] s3Objects => {
-            io:println("Listing all object: ");
-            foreach s3Object in s3Objects {
-                io:println("---------------------------------");
-                io:println("Object Name: " + s3Object.objectName);
-                io:println("Object Size: " + s3Object.objectSize);
-            }
+    if (getAllObjectsResponse is amazons3:S3Object[]) {
+        io:println("Listing all object: ");
+        foreach s3Object in getAllObjectsResponse {
```
Comment (ldclakmal) [inline amazons3/Module.md]:
```suggestion
        foreach var s3Object in getAllObjectsResponse {
```

File: amazons3/amazons3_endpoint.bal [hunk 6]
```diff
@@ -19,12 +19,249 @@
...
+    # Delete an object.
+    # + objectName - The name of the object
+    # + return - If success, returns Status object, else returns error
+    remote function deleteObject(string bucketName, string objectName) returns Status|error;
+
+    # Delete a bucket.
+    # + return - If success, returns Status object, else returns error
+    remote function deleteBucket(string bucketName) returns Status|error;
+};
+
```
Comment (ldclakmal) [inline amazons3/amazons3_endpoint.bal]: Shall we remove the new line?

File: amazons3/amazons3_endpoint.bal [hunk 7]
```diff
@@ -19,12 +19,249 @@
...
+    var httpResponse = self.amazonS3Client->get(requestURI, message = request);
+    if (httpResponse is http:Response) {
+        int statusCode = httpResponse.statusCode;
+        var amazonResponse = httpResponse.getXmlPayload();
+        if (amazonResponse is xml) {
+            if (statusCode == 200) {
+                return getS3ObjectsList(amazonResponse);
+            }
+            else{
```
Comment (ldclakmal) [inline amazons3/amazons3_endpoint.bal]: Shall we move this to above line?
```suggestion
            else {
```

File: amazons3/amazons3_endpoint.bal [hunk 8]
```diff
@@ -19,12 +19,249 @@
...
+
+    var httpResponse = self.amazonS3Client->get(requestURI, message = request);
+    if (httpResponse is http:Response) {
+        int statusCode = httpResponse.statusCode;
+        var amazonResponse = httpResponse.getPayloadAsString();
+        if (amazonResponse is string) {
+            if (statusCode == 200) {
+                return getS3Object(amazonResponse);
+            }
+            else{
```
Comment (ldclakmal) [inline amazons3/amazons3_endpoint.bal]: Shall we move this to above line?
```suggestion
            else {
```

File: amazons3/amazons3_endpoint.bal [hunk 9]
```diff
@@ -19,12 +19,249 @@
...
+                error err = error(<string>statusCode, { message : "Error occurred while getting the amazonS3 object." });
+                return err;
+            }
+        } else {
+        error err = error(AMAZONS3_ERROR_CODE, { message : "Error occurred while accessing the string payload
```
Comment (ldclakmal) [inline amazons3/amazons3_endpoint.bal]: Shall we format the code here?

File: amazons3/amazons3_endpoint.bal (hunk 6 above)
Comment (kalaiyarasiganeshalingam) [reply amazons3/amazons3_endpoint.bal]: Fixed.

**Pull request merged**